### PR TITLE
Module Testing, Server Module Initialisations

### PR DESCRIPTION
**Pull request by bruise-88**

### CODE REVIEW

File: Dashboard/Client/SessionManagement/ClientSessionManager.cs [hunk 1]
```diff
@@ -29,9 +32,10 @@ public ClientSessionManager()
             _serializer = new Serializer();
             _communicator = CommunicationFactory.GetCommunicator();
             _communicator.Subscribe(moduleIdentifier, this);
+            _contentClient = ContentClientFactory.getInstance();
+
             TraceManager session = new();
-            session.TraceListener();
-           
+            session.TraceListener();
```
Comment (Yogesh7920) [inline Dashboard/Client/SessionManagement/ClientSessionManager.cs:38]: Add trace before initializing other modules. and also initialize whiteboard and screen share.

Comment (bruise-88) [reply Dashboard/Client/SessionManagement/ClientSessionManager.cs:38]: Initialisations on client side?

Comment (Yogesh7920) [reply Dashboard/Client/SessionManagement/ClientSessionManager.cs:38]: yup

**Comment by Yogesh7920** [comment]
Create the function stop to call stop of communicator

File: Dashboard/Client/SessionManagement/ClientSessionManager.cs [hunk 2]
```diff
@@ -50,12 +55,13 @@ public ClientSessionManager()
...
         public ClientSessionManager(ICommunicator communicator)
         {
+            TraceManager session = new();
             moduleIdentifier = "Dashboard";
             _serializer = new Serializer();
             _communicator = communicator;
             _communicator.Subscribe(moduleIdentifier, this);
-            TraceManager session = new();
-            session.TraceListener();
+            clientBoardStateManager = ClientBoardStateManager.Instance;
```
Comment (Yogesh7920) [inline Dashboard/Client/SessionManagement/ClientSessionManager.cs:63]: Why use the actual whiteboard for testing. 
u can remove this whiteboard state manager here.

Comment (bruise-88) [reply Dashboard/Client/SessionManagement/ClientSessionManager.cs:63]: That I am not using in testing

Comment (bruise-88) [reply Dashboard/Client/SessionManagement/ClientSessionManager.cs:63]: When I will use, I will create a fake for it

**Pull request merged**